src/psyclone/tests/parse/module_manager_test.py

```diff
@@ -0,0 +1,314 @@
+        ModuleManager()
+
+    assert ("You need to use 'ModuleManager.get()' to get the singleton "
+            "instance." in str(err.value))
+
+
+# ----------------------------------------------------------------------------
+def test_mod_manager_get_modules_in_file():
+    '''Tests that file names are mapped as expected to the module they
+    contain. '''
+    mod_man = ModuleManager.get()
+    assert mod_man.get_modules_in_file("a_mod.f90") == ["a_mod"]
+    assert mod_man.get_modules_in_file("b_mod.F90") == ["b_mod"]
+    assert mod_man.get_modules_in_file("c_mod.x90") == ["c_mod"]
+    assert mod_man.get_modules_in_file("d_mod.X90") == ["d_mod"]
+    assert mod_man.get_modules_in_file("d.f90") == []
+
+
+# ----------------------------------------------------------------------------
+def mod_man_test_setup_directories():
+    '''Sets up a directory and file structure for several of the following
+    tests. The following structure is created - note that each Fortran file
+    declares a module of the same name as the basename of the file:
+    tmp/d1/a_mod.f90       : no dependencies
+    tmp/d1/d3/b_mod.F90    : no dependencies
+    tmp/d1/d3/c_mod.x90    : depends on a_mod/b_mod
+    tmp/d2/d_mod.X90       : depends on c_mod
+    tmp/d2/d4/e_mod.F90    : depends on netcdf
+    tmp/d2/d4/f_mod.ignore
+    '''
+
+    os.makedirs("d1/d3")
+    os.makedirs("d2/d4")
+    for (name, path, dependencies) in [("a_mod.f90", "d1", []),
+                                       ("b_mod.F90", "d1/d3", []),
+                                       ("c_mod.x90", "d1/d3", ["a_mod",
+                                                               "b_mod"]),
+                                       ("d_mod.X90", "d2", ["c_mod"]),
+                                       ("e_mod.F90", "d2/d4", ["netcdf"]),
+                                       ("f_mod.ignore", "d2/d4", [])]:
+        # Create a list of "use a_mod, only: a_mod_symbol" statements
+        uses = "\n".join(f"use {dep}, only: {dep}_symbol"
+                         for dep in dependencies)
+        base, _ = os.path.splitext(name)
+        with open(os.path.join(path, name), "w", encoding="utf-8") as f_out:
+            f_out.write(f"module {base}\n{uses}\nend module {base}")
+
+
+# ----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir")
+def test_mod_manager_directory_reading():
+    '''Tests that directories are read as expected. We use the standard
+    directory and file setup (see mod_man_test_setup_directories).
+    tmp/d1/a_mod.f90
+    tmp/d1/d3/b_mod.F90
+    tmp/d1/d3/c_mod.x90
+    tmp/d2/d_mod.X90
+    tmp/d2/d4/e_mod.F90
+    tmp/d2/d4/f_mod.ignore
+    '''
+
+    mod_man_test_setup_directories()
+    mod_man = ModuleManager.get()
+
+    # Add a path to the directory recursively (as default):
+    mod_man.add_search_path("d1")
+    assert mod_man._search_paths == ["d1", "d1/d3"]
```

Comment:
The paths come from os.walk, so I would assume they are OS independent. It just means that this test won't work natively in windows (since it would have a drive or so???), but I don't hard-code a '/' (except for in the test). Or do you prefer the test to use `os.path.join`?

Reply:
No, it's all fine - this of course is just test output so I'm happy.